streams/src/main/java/org/apache/kafka/streams/kstream/Consumed.java

```diff
@@ -55,9 +56,12 @@ public class Consumed<K, V> implements NamedOperation<Consumed<K, V>> {
     protected Serde<K> keySerde;
     protected Serde<V> valueSerde;
     protected TimestampExtractor timestampExtractor;
-    protected Topology.AutoOffsetReset resetPolicy;
+    protected AutoOffsetReset resetPolicy;
+    @Deprecated
+    protected Topology.AutoOffsetReset resetPolicy; // Replaced with new AutoOffsetReset class introduced in 4.0. 
     protected String processorName;
 
+    @Deprecated
```

Comment:
No need to `@Deprecate`a `private` constructor -- it's not user facing (also c.f my comment below -- we should only have once constructor.

Reply:
Thank you, removed deprecated comment and removed the second constructor. We will now just have the updated one that you mentioned in the below comment.